### PR TITLE
Build + LibTest + Tests: Convert more tests to be LibTest based. 

### PR DESCRIPTION
**Pull request by bgianfo**
More progress towards standardizing on a test framework for userland tests (#4078)

- __Build: Remove unused ${REGEX_SOURCES} from the tests CMakeLists.txt__

- __Tests: Rename LibGfx painter benchmark to denote it's not a test.__

-  __Tests: Convert LibGfx image decoder test to be LibTest based.__

- __Tests: Convert LibGfx font handling test to be LibTest based.__

- __Build: Simplify LibGfx test's CMakeLists.txt with serenity_test(..)__

  Now that everything is LibTest based, we can just use the normal
  pattern of iterating + serenity_test(..).

- __Tests: Unify LibC tests to single location.__

  In a1720eed2a8e744f2cd6d99adae4dff1fbab309e I added this new test,
  but missed that there were already some "unit tests" for LibC over
  in Userland/Tests/LibC. So lets unify these two locations.

- __Tests: Enable Spreadsheet unit tests so they actually build__

- __LibTest: Add EXPECT_NE(..) test assertion.__

- __Tests: Convert LibC mktemp tests to be LibTest based.__

- __Tests: Convert LibC exec unit test to be LibTest based.__

### CODE REVIEW

**Comment by awesomekling** [comment]
Please remove all the `Signed-off-by: Myself` lines :)

**Comment by bgianfo** [comment]
> Please remove all the `Signed-off-by: Myself` lines :)

Not down with the developer certificate of origin?
Sure.

**Comment by bgianfo** [comment]
done

File: Userland/Tests/LibC/TestLibCExec.cpp [hunk 1]
```diff
@@ -4,23 +4,22 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include <LibTest/TestCase.h>
 #include <fcntl.h>
 #include <stdio.h>
 #include <unistd.h>
 
-int main()
+TEST_CASE(exec_should_not_serach_current_directory)
```
Comment (linusg) [inline Userland/Tests/LibC/TestLibCExec.cpp:12]:
```suggestion
TEST_CASE(exec_should_not_search_current_directory)
```

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Tests/LibC/TestLibCExec.cpp (hunk 1 above)
Comment (bgianfo) [reply Userland/Tests/LibC/TestLibCExec.cpp]: Thanks, fixed!

**Pull request merged**